src/Lecture3.hs

```diff
@@ -134,11 +152,17 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
-
+  (<>) :: List1 a -> List1 a -> List1 a
+  (List1 x xs) <> (List1 _ ys) = List1 x (xs ++ ys)
```

Comment:
I see that you ignore the first element of the second list. It's interesting that this implementation passes tests 🙂 But I believe you should use all elements of both lists when you append them.
```suggestion
  (List1 x xs) <> (List1 y ys) = List1 x (xs ++ ys)
```

Reply:
@chshersh I am confused -- even your suggested change ignores the `y` of second list. What am I missing?

Reply:
@kpadmasola I haven't implemented the function completely 😉 
You need to use `y` in there. So yeah, my suggestion is still invalid. But I believe you can finish it! 💪🏻